### ISSUE TITLE
Is it possible to restore global_step in keras?

### ISSUE THREAD

**Issue by scotthuang1989** [keras-team/keras#9546]
keras can save checkpoint with ModelCheckpoint, then restore weights from checkpoint with load_weights,

Here is a scenario,
1. train a model 1000 epoch. save the weights to checkpoint file.
2. restore weight from checkpoint file, train another 1000 epochs.
3. In tensorflow the starting global step of 2nd training is 10001, but in keras, it seems that it reset the global steps(or, no such thing?).

So far, this behavior cause me 2 problems.

1. tensorboad is messing up, the 2 training run all start from step 0.

![default](https://user-images.githubusercontent.com/5325686/36935897-9a76e428-1f38-11e8-947e-5d513df0b1b9.png)

2.  I need save some information into file during the training, the file name will depends on the train step, so I can save information for different steps, eg: weight_1000steps.h5, weight_2000steps.h5, etc.
But since step will restart if I start a new training, so some old file will be overwritten.

**my question is:**

How can I restore the training step/epoch from chekpoint file in keras?

**Comment by hadaev8** [comment]
I would like to know too

**Comment by scotthuang1989** [comment]
@hadaev8 , if you witch x-axis from steps to time. then tensorboard will work fine.

**Comment by alexwal** [comment]
Not if there are long gaps between when you stop and resume training

**Comment by CHIKKIZZY** [comment]
Is there a solution to the issue raised here?

**Comment by whhu** [comment]
A custom callback is implemented to solve this problem at [stackoverflow](https://stackoverflow.com/a/66389445/8163786).
The callback initializes the private counter of the model with the one of the optimizer `on_train_begin`.
